### PR TITLE
Add CLI benchmark device selector

### PR DESCRIPTION
**Pull request by AdamHarries**
At present, there is no offical way in the SYCL spec of selecting a specific OpenCL device for an existing, compiled application. This PR provides a SYCL device selector for matching devices of a requested vendor and type, as well as a command line flag, `--device` for the benchmark applications for triggering the selector.

### CODE REVIEW

File: benchmarks/benchmark.h [hunk 1]
```diff
@@ -95,6 +106,36 @@ struct benchmark_arguments {
...
+          std::cerr << " Incorrect parameter " << std::endl;
+          match = false;
+          break;
+        }
+        std::string outputOption = argv[i + 1];
+        std::transform(outputOption.begin(), outputOption.end(),
+                       outputOption.begin(), ::tolower);
+        // split the string into tokens on ':'
+        std::stringstream ss;
+        ss.str(outputOption);
```
Comment (Ruyk) [inline benchmarks/benchmark.h]: why don't use directly the constructor that takes an string?

File: benchmarks/sycl_exclusive_scan.cpp [hunk 2]
```diff
@@ -41,25 +41,27 @@ using namespace sycl::helpers;
...
 
   auto exclusive_scan = [&]() {
-    cl::sycl::queue q;
+    cl::sycl::queue q(cds);
     sycl::sycl_execution_policy<class ExclusiveScanAlgorithm1> snp(q);
-    std::experimental::parallel::exclusive_scan(snp, begin(v1), end(v1), 
-      begin(v1), 0, [=](int x, int y){ return x + y; });
+    std::experimental::parallel::exclusive_scan(
+        snp, begin(v1), end(v1), begin(v1), 0,
+        [=](int x, int y) { return x + y; });
```
Comment (Ruyk) [inline benchmarks/sycl_exclusive_scan.cpp]: is this clang-format?

File: benchmarks/benchmark.h [hunk 3]
```diff
@@ -61,6 +69,9 @@ struct benchmark_arguments {
         validProgramOptions(true) {
     /* Match parameters */
     std::regex output_regex("--output");
+    std::regex device_regex("--device");
+    // device_vendor = std::string(".*");
+    // device_type   = std::string("*");
```
Comment (AdamHarries) [inline benchmarks/benchmark.h]: Ah, that was some changes I made while working out how to handle wildcards, and where the `--device` specifier isn't given. I'll remove them.

File: benchmarks/benchmark.h [hunk 4]
```diff
@@ -95,6 +106,36 @@ struct benchmark_arguments {
         i++;
       }
 
+      // Check for the --device parameter
+      if (std::regex_match(option, device_regex)) {
+        if (i + 1 >= argc) {
```
Comment (AdamHarries) [inline benchmarks/benchmark.h]: WIll do.

File: benchmarks/sycl_exclusive_scan.cpp (hunk 2 above)
Comment (AdamHarries) [inline benchmarks/sycl_exclusive_scan.cpp]: I think it probably is - I ran clang-format over all the files that I'd changed before committing.

File: benchmarks/benchmark.h (hunk 1 above)
Comment (AdamHarries) [inline benchmarks/benchmark.h]: No reason!

**Pull request merged**